tests/models/test_data.py

```diff
@@ -72,41 +95,138 @@ def __init__(self):
-            f'x: {x_np.shape[0]}, y: {y_np.shape[0]}, latlons: {latlons.shape[0]}'
+            assert (
+                type(x_train_data.historical) == np.ndarray
+            ) and (type(y_np) == np.ndarray)
+
+        if (not normalize) and (experiment == 'nowcast') and (not to_tensor):
+            assert (
+                x_train_data.historical.shape[0] == x_train_data.current.shape[0]
+            ), "The 0th dimension (latlons) should be equal in the " \
+                f"historical ({x_train_data.historical.shape[0]}) and " \
+                f"current ({x_train_data.current.shape[0]}) arrays."
+
+            expected = (
+                x.precip
+                .sel(time=y.time)
+                .stack(dims=['lat', 'lon'])
+                .values.T
+            )
+            got = x_train_data.current
+            if surrounding_pixels is None:
+                assert expected.shape == got.shape, "should have stacked latlon" \
+                    " vars as the first dimension in the current array."
+
+                assert all(expected == got), "" \
+                    "Expected to find the target timesetep of `precip` values"\
+                    "(the non-target variable for the target timestep: " \
+                    f"({pd.to_datetime(y.time.values).strftime('%Y-%m-%d')[0]})." \
+                    f"Expected: {expected[:5]}. Got: {got[:5]}"
+
+        if normalize and (experiment == 'nowcast') and (not to_tensor):
+            assert x_train_data.current.max() < 6, f"The current data should be" \
+                f" normalized. Currently: {x_train_data.current.flatten()}"
 
         for idx in range(latlons.shape[0]):
-
             lat, lon = latlons[idx, 0], latlons[idx, 1]
-
-            for time in range(x_np.shape[1]):
+            for time in range(x_train_data.historical.shape[1]):
                 target = x.isel(time=time).sel(lat=lat).sel(lon=lon).VHI.values
 
                 if (not normalize) and (not to_tensor):
-                    assert target == x_np[idx, time, 0], \
-                        f'Got different x values for time idx: {time}, lat: {lat}, ' \
-                        f'lon: {lon}.Expected {target}, got {x_np[idx, time, 0]}'
-
-            if not to_tensor:
-                target_y = y.isel(time=0).sel(lat=lat).sel(lon=lon).VHI.values
-                assert target_y == y_np[idx, 0], \
-                    f'Got y different values for lat: {lat}, ' \
-                    f'lon: {lon}.Expected {target_y}, got {y_np[idx, 0]}'
+                    assert target == x_train_data.historical[idx, time, 0], \
+                        'Got different x values for time idx:'\
+                        f'{time}, lat: {lat}, lon: {lon} Expected {target}, '\
+                        f'got {x_train_data.historical[idx, time, 0]}'
+
+    @pytest.mark.parametrize('normalize', [True, False])
+    def test_ds_to_np_multi_vars(self, tmp_path, normalize):
```

Comment:
I think this could be absorbed into the test above. Is anything tested in the single variable test that isn't tested in the multi variable case?

Reply:
I think it's important to have a multi-variable test case for 2 reasons
1) we will be using multi variable experiments most of the time
2) the dimensions of the multi-variable arrays will be different and can we not imagine a scenario where the tests pass for the single variable case but fail for the multi-variable case?

Reply:
ohhhh you mean to incorporate into the tests above. Good point!

Reply:
okay done!